tests/BonsaiStarter.t.sol

```diff
@@ -14,51 +14,44 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-pragma solidity ^0.8.17;
+pragma solidity ^0.8.19;
 
 import {BonsaiTest} from "bonsai/BonsaiTest.sol";
-import {IBonsaiRelay} from "bonsai/IBonsaiRelay.sol";
-import {BonsaiStarter} from "contracts/BonsaiStarter.sol";
-
-contract BonsaiStarterTest is BonsaiTest {
-    function setUp() public withRelay {}
-
-    // Test the BonsaiStarter contract by mocking an off-chain callback request
-    function testOffChainMock() public {
-        bytes32 imageId = queryImageId("FIBONACCI");
-        // Deploy a new starter instance
-        BonsaiStarter starter = new BonsaiStarter(IBonsaiRelay(bonsaiRelay), imageId);
-
-        // Anticipate a callback invocation on the starter contract
-        vm.expectCall(address(starter), abi.encodeWithSelector(BonsaiStarter.storeResult.selector));
-        // Relay the solution as a callback
-        uint64 BONSAI_CALLBACK_GAS_LIMIT = 100000;
-        runCallbackRequest(
-            imageId, abi.encode(128), address(starter), starter.storeResult.selector, BONSAI_CALLBACK_GAS_LIMIT
-        );
-
-        // Validate the Fibonacci solution value
-        uint256 result = starter.fibonacci(128);
-        assertEq(result, uint256(407305795904080553832073954));
+import {console2} from "forge-std/console2.sol";
+import {IRiscZeroVerifier} from "bonsai/IRiscZeroVerifier.sol";
+import {ControlID, RiscZeroGroth16Verifier} from "bonsai/groth16/RiscZeroGroth16Verifier.sol";
+import {RiscZeroGroth16VerifierTest} from "./RiscZeroGroth16VerifierTest.sol";
+import "../contracts/BonsaiStarter.sol";
+
+contract EvenNumberTest is BonsaiTest {
+    EvenNumber public evenNumber;
+
+    function setUp() public {
+        bytes32 imageId = queryImageId("IS_EVEN");
+        IRiscZeroVerifier verifier = deployRiscZeroGroth16Verifier();
+        evenNumber = new EvenNumber(verifier, imageId);
+        assertEq(evenNumber.get(), 0);
     }
 
-    // Test the BonsaiStarter contract by mocking an on-chain callback request
-    function testOnChainMock() public {
-        // Deploy a new starter instance
-        BonsaiStarter starter = new BonsaiStarter(IBonsaiRelay(bonsaiRelay), queryImageId("FIBONACCI"));
-
-        // Anticipate an on-chain callback request to the relay
-        vm.expectCall(address(bonsaiRelay), abi.encodeWithSelector(IBonsaiRelay.requestCallback.selector));
-        // Request the on-chain callback
-        starter.calculateFibonacci(128);
-
-        // Anticipate a callback invocation on the starter contract
-        vm.expectCall(address(starter), abi.encodeWithSelector(BonsaiStarter.storeResult.selector));
-        // Relay the solution as a callback
-        runPendingCallbackRequest();
+    function testSet() public {
+        uint256 number = 12345678;
+        bytes32 imageId = queryImageId("IS_EVEN");
```

Comment:
This runs `cargo` a second time, which add significant time to the test. It would be better to store the image ID when it is first fetched as part of `setUp()`